src/components/RouteOption/DockedEbikeRouteOption.jsx

```diff
@@ -34,22 +35,30 @@ export default function DockedEbikeRouteOption({
       <header>
         <h1>{routeOption.provider.name}</h1>
       </header>
+
       {fromDockingStations.length > 0 && toDockingStations.length > 0 ? (
-        <div style={{ display: "flex" }}>
-          <DockingStationSelector
-            label="From docking station"
-            value={fromDockingStationId}
-            stations={fromDockingStations}
-            onChange={onStartingPointStationChange}
-            style={{ flex: 1 }}
-          />
-          <DockingStationSelector
-            label="To docking station"
-            value={toDockingStationId}
-            stations={toDockingStations}
-            onChange={onDestinationStationChange}
-            style={{ flex: 1 }}
-          />
+        <div>
+          <div style={{ display: "flex" }}>
+            <DockingStationSelector
+              label="From docking station"
+              value={fromDockingStationId}
+              stations={fromDockingStations}
+              onChange={onStartingPointStationChange}
+              style={{ flex: 1 }}
+            />
+            <DockingStationSelector
+              label="To docking station"
+              value={toDockingStationId}
+              stations={toDockingStations}
+              onChange={onDestinationStationChange}
+              style={{ flex: 1 }}
+            />
+          </div>
+
+          {/* Render the details only if routeOption.details is not null and the option is selected */}
```

Comment:
this comment does not add much value IMO - it describes precisely what the next line does. please consider removing this comment